src/status_im/contexts/wallet/swap/events.cljs

```diff
@@ -369,3 +369,29 @@
                                      :receive-token-symbol receive-token-symbol
                                      :receive-amount       receive-amount})
                         (i18n/label :t/swap-failed))}]]]}))))
+
+(rf/reg-event-fx :wallet.swap/flip-assets
+ (fn [{:keys [db]}]
+   (let [{:keys [asset-to-pay asset-to-receive
+                 swap-proposal amount]} (get-in db [:wallet :ui :swap])
+         receive-token-decimals         (:decimals asset-to-receive)
+         amount-out                     (when swap-proposal (:amount-out swap-proposal))
+         receive-amount                 (when amount-out
+                                          (number/remove-trailing-zeroes
```

Comment:
A couple comments about this part :)

1. This is quite hard to read from the inside out. What do you think of using the thread first macro?

```clojure
(when amount-out
  (-> amount-out
      hex/normalize-hex
      native-module/hex-to-number
      (number/convert-to-whole-number receive-token-decimals)
      (number/to-fixed receive-token-decimals)))
```

2. Notice that I used `number/to-fixed`. I think we should avoid `toFixed` because it throws an exception on nil values. We never want unhandled exceptions to happen on the client because this will break the UX.

3. There are too many calls that a dev needs to remember to carefully call in order. There is another place which is similar https://github.com/status-im/status-mobile/blob/783504f33261e229a79ad6aa65fa8ecc7334ee93/src/status_im/contexts/wallet/swap/setup_swap/view.cljs#L171-L180

Better I think would be to have two separate utility functions. Both handle (ignore) nil. One could be called `number/to-fixed` and internally use `remove-trailing-zeroes` and another could be `number/hex->whole`, which would call the whole chain `normalize-hex` + `hex-to-number` + `convert-to-whole-number`.

Reply:
Nice suggestions @ilmotta, using thread first macro makes the code much cleaner, I will implement them.